crates/storage/provider/src/providers/snapshot/manager.rs

```diff
@@ -452,21 +458,40 @@ impl SnapshotProvider {
         let mut provider = get_provider(range.start)?;
         let mut cursor = provider.cursor()?;
 
+        // Used as a safety check to make sure we don't get stuck on a loop on an unexpected issue
+        // or corrupted file.
+        //
+        // Since we're looping through a range, we should retry at most once for a specific
+        // **number: if we're changing files (get_provider) aka checking the lower block range file.
+        let mut retrying = false;
```

Comment:
should this be in the 'outer loop?

because this is persisted across `number`

Reply:
shouldnt be an issue, but moved it inside since yeah it makes more sense